### ISSUE TITLE
ChatHub should sync chat histories with chat bots.

### ISSUE THREAD

**Issue by DrewNeilson** [chathub-dev/chathub#704]
I've noticed that if in ChatHub I have a conversation with Bing Chat or with Google Bard, that conversation subsequently appears in the "recent activity" section on Bing Chat's official web page, or in the "recent" section on Google Bard's official web page. How does this happen?
Unfortunately, the communication is only one way. ChatHub does not retrieve from Bing Chat nor from Bard any conversations that I had on their respective web pages, nor does it retrieve from those services any updates on my conversations that I've started in ChatHub and continued on the respective web page. If ChatHub and Bing Chat would sync my Bing Chat conversations, and if ChatHub and Bard would sync my Bard conversations, then I could start a conversation with one of those two chatbots on my PC, and continue that conversation on my iPhone while I'm away from my PC by opening the Bing mobile app and clicking on that conversation in my history, or by opening Bard's web page and doing the same thing. When I return to my PC, I could open ChatHub, and continue the conversation. Throughout this process, the conversation would be a single conversation, synchronized between ChatHub and Bing Chat or Google Bard.

**Comment by wong2** [comment]
The ChatHub conversation was "synced" to Bing or Bard because it's calling API of these services to create conversation and chat.
But the other way round would be much harder to implement.

**Comment by DrewNeilson** [comment]
> the other way round would be much harder to implement.

One of my biggest requests, if not my biggest request, is that this be implemented. ChatHub is useful, but it's less useful than it could be because it does not see conversation activity that occurred outside of ChatHub. I'm considering buying a Premium license for the ability to search conversations for specific text, but one thing that's holding me back is the knowledge that any chatbot conversations that I've had outside of ChatHub, such as the many conversations--probably about 20--before I started using ChatHub, and all chatbot conversations that, going forward, I have on my phone are invisible to ChatHub and thus are not searchable. Even if when I'm on my PC I only interact with chatbots in ChatHub, the conversations that I have with those chatbots when I'm on my phone--which mostly happens when I'm away from my PC--are invisible to ChatHub.

**Comment by wong2** [comment]
I agree that would be perfect. But it's just tooo hard to implement.

**Comment by DrewNeilson** [comment]
Do any of the chatbots that ChatHub supports--especially Bing and Bard--have APIs that support 2-way synchronization?
